api/python/quilt3/packages.py

```diff
@@ -622,6 +623,10 @@ def set_dir(self, lkey, path=None, meta=None):
             for obj in objects:
                 if not obj['IsLatest']:
                     continue
+                # Skip S3 pseduo directory files
+                if obj['Key'].endswith('/') and obj['Size'] == 0:
+                    warnings.warn(f'Logical keys cannot end in "/", skipping: {obj["Key"]}')
```

Comment:
When does this actually happen? If regular nested directories in s3 show up here, we should skip them silently; a warning doesn't really help the user.